src/project/git/GitCommandGitProject.ts

```diff
@@ -69,11 +78,16 @@ export class GitCommandGitProject extends NodeFsLocalProject implements GitProje
                          directoryManager: DirectoryManager = DefaultDirectoryManager): Promise<GitCommandGitProject> {
         return clone(credentials, id, opts, directoryManager)
             .then(p => {
-                const pathIntoRepo = !!id.path ?
-                    id.path.startsWith("/") ? id.path : "/" + id.path :
-                    "";
-                const gp = GitCommandGitProject.fromBaseDir(id, p.baseDir + pathIntoRepo, credentials);
-                return gp;
+                if (!!id.path) {
+                    const pathInsideRepo = id.path.startsWith("/") ? id.path : "/" + id.path;
+                    // not sure this will work with cached
+                    const gp = GitCommandGitProject.fromBaseDir(id, p.baseDir + pathInsideRepo, credentials,
+                        () => p.release(),
+                        p.provenance + "\ncopied into one with extra path " + id.path);
+                    return gp;
+                } else {
+                    return p;
```

Comment:
the common case is: clone() now returns a GitCommandGitProject

Reply:
Why not just `GitProject`? This is encouraging callers to couple more deeply. What is the motivation for this change?